#!/usr/bin/env python
PACKAGE = "commbot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# cue types
gen.add("lights_traj", bool_t, 0, "Lights cue trajectory mode", False)
gen.add("lights_goal", bool_t, 0, "Lights cue goal mode", False)
gen.add("projector_traj", bool_t, 0, "Projector cue trajectory mode", False)
gen.add("projector_goal", bool_t, 0, "Projector cue goal mode", False)

# common
gen.add("length_traj", double_t, 0, "Distance along trajectory used to animate trajectory cues.", 1.0, 0.5, 10)
gen.add("k_freq_goal", double_t, 0, "Coefficient on 1/prox used to calculate freq.", 6, 0.1, 10)
gen.add("max_freq", double_t, 0, "", 5, 0.5, 10)

# lights
gen.add("k_lights_freq_traj", double_t, 0, "Coefficient for calculating frequency from forthcoming turn angle.", 0.1, 0.01, 1.0)
gen.add("lights_max_angle_straight", double_t, 0, "Max angle for a turn to be considered 'straight'.", 20.0, 0.0, 180.0)

# projector goal arrow
gen.add("projector_goal_arrow_length", double_t, 0, "Length of goal arrow.", 4.0, 0.1, 10.0)
gen.add("projector_goal_ratio_arrow_length_total_length", double_t, 0, "Ratio between arrow length and total length", 0.25, 0.05, 1)
gen.add("projector_goal_ratio_dashes_width_total_length", double_t, 0, "Ratio between dashes width and total length", 0.25, 0.1, 1)
gen.add("projector_goal_ratio_arrow_width_dashes_width", double_t, 0, "Ratio between arrow width and dashes width", 2.0, 1, 3)
gen.add("projector_goal_shift_x", double_t, 0, "", 0.2, -10, 10)
gen.add("projector_goal_shift_y", double_t, 0, "", -1.0, -10, 10)
gen.add("projector_prox_origin_min", double_t, 0, "", -2.2, -10.0, 10.0)
gen.add("projector_prox_origin_max", double_t, 0, "", 5.2, -10.0, 10.0)

# projector traj arrow
gen.add("projector_traj_arrow_length", double_t, 0, "Length of trajectory arrow.", 4.0, 0.1, 10.0)
gen.add("projector_traj_ratio_shaft_width_arrow_length", double_t, 0, "Ratio between shaft width and total length.", 0.25, 0.1, 1)
gen.add("projector_traj_ratio_head_width_shaft_width", double_t, 0, "Ratio between head width and shaft width.", 2.0, 1, 5.0)
gen.add("projector_traj_ratio_head_length_arrow_length", double_t, 0, "Ratio between head length and total length.", 0.5, 0.1, 1)
gen.add("projector_traj_shift_x", double_t, 0, "", 2.8, -10, 10)
gen.add("projector_traj_shift_y", double_t, 0, "", 1.2, -10, 10)

exit(gen.generate(PACKAGE, "commbot", "Commbot"))
